src/c++/perf_analyzer/request_rate_manager.cc

```diff
@@ -130,18 +130,21 @@ RequestRateManager::CreateWorkerSchedules(
 
   std::vector<RateSchedulePtr_t> worker_schedules =
       CreateEmptyWorkerSchedules();
+  std::vector<size_t> thread_ids{CalculateThreadIds()};
 
   std::chrono::nanoseconds next_timestamp(0);
+  size_t thread_id_index = 0;
   size_t worker_index = 0;
 
+
   // Generate schedule until we hit max_duration, but also make sure that all
-  // worker schedules are the same length by continuing until worker_index is
-  // back to 0
+  // worker schedules follow the thread id distribution
   //
-  while (next_timestamp < max_duration || worker_index != 0) {
+  while (next_timestamp < max_duration ||
+         thread_id_index % thread_ids.size() != 0) {
     next_timestamp = next_timestamp + distribution(schedule_rng);
+    worker_index = (thread_id_index++) % workers_.size();
```

Comment:
I'm not convinced this is working correctly (since you aren't actually using the vector you generated), and I don't see any testing to show that it would be working.
You should be able to test the actual generated worker schedules.  For the case of 5 seq in 4 threads, I would expect thread0 to have twice as big of a schedule as the others. I don't believe that would happen with this code.

Reply:
Or, more ideally, write a test that shows that every sequence is issued the same number of times.

Reply:
Actually, you can't test this invariant yet without my story:
> every sequence is issued the same number of times.

But you can and should test this:
>  For the case of 5 seq in 4 threads, I would expect thread0 to have twice as big of a schedule as the others.

Reply:
There should be new testing to verify the ratio of schedule sizing now.